xnmt/translator.py

```diff
@@ -277,9 +307,16 @@ def __call__(self, translator, dec_state, src, trg):
       except ValueError:
         pass
       # Calculate the evaluation score
-      eval_score.append(self.evaluation_metric.evaluate_fast(trg_i.words, sample_i))
+      score = self.evaluation_metric.evaluate_fast(trg_i.words, sample_i)
+      eval_score.append(0 if math.isnan(score) else score)
```

Comment:
Do not squash nan to 0, this may create a more severe bug in the future I think.

Reply:
Makes sense. Currently, this nan is thrown when the target sentence is of zero length. I can test that before.